manifests/extpack.pp

```diff
@@ -0,0 +1,88 @@
+# == Class: virtualbox::extpack
+#
+# This class (un)installs Oracle's VirtualBox extension pack.
+#
+# === Parameters
+#
+# [*ensure*]
+#   Set to 'present' to install extension pack. Set to 'absent' to uninstall.
+#   Defaults to 'present'
+#
+# [*version*]
+#   Full version of the Extension Pack to install. E.g. '4.3.20r96996'. This
+#   parameter is mandatory.
+#
+# [*source*]
+#   Download extension pack from the given URL. If not specified download from
+#   Oracle's official server.
+#
+class virtualbox::extpack (
+  $ensure  = $virtualbox::params::extpack_ensure,
+  $version = $virtualbox::params::extpack_version,
+  $source  = $virtualbox::params::extpack_source
+) inherits virtualbox::params {
+  validate_string($ensure)
+  validate_string($version)
+  validate_string($source)
```

Comment:
This effectively makes 'source' required. If it's undef, it's not a string. We should change this to read:

```
if $source {
  validate_string($source)
}
```